src/core/memory.d

```diff
@@ -893,7 +893,7 @@ void pureFree(void* ptr) @system pure @nogc nothrow
 
     // subtract 2 because snn.lib adds 2 unconditionally before passing
     //  the size to the Windows API
-    void* z = pureMalloc(size_t.max - 2); // won't affect `errno`
+    void* z = pureMalloc(long.max - 2); // won't affect `errno`
```

Comment:
This probably would be ptrdiff_t for the benefit of 32bit systems? I don't recall the signature of gc.malloc within druntime.

Reply:
In any case, the bug applies only when you call malloc with a size from `size_t.max-23ʼ and up.

Shifting the value by one bit would be sufficient also (`size_t.max >> 1`)

Reply:
Or unsetting the right most 8 bits to keep the adjustment small (`size_t.max & ~255`).

Its all just a workaround of a libc bug anyway. ☺️

Reply:
> Or unsetting the right most 8 bits to keep the adjustment small (size_t.max & ~255).

Seems like that worked fine :)